src/nano_contracts/types.ts

```diff
@@ -4,18 +4,94 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
+import { z } from 'zod';
```

Comment:
Changing the types to be derived from a zod schema was meant to allow parsing and checking the types more easily where needed.